### PR TITLE
[ticket/13162] Add truncate table functionality to database tools

### PR DESCRIPTION
**Pull request by marc1706**
Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/master/coding-guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket:

https://tracker.phpbb.com/browse/PHPBB3-13162

### CODE REVIEW

**Comment by rubencm** [comment]
https://github.com/phpbb/phpbb/pull/5329/files

File: phpBB/phpbb/db/tools/doctrine.php [hunk 1]
```diff
@@ -363,6 +363,21 @@ function (Schema $schema) use ($table_name, $column): void
...
+	 */
+	public function sql_truncate_table(string $table_name): void
+	{
+		try
+		{
+			$this->connection->executeQuery($this->get_schema_manager()->getDatabasePlatform()->getTruncateTableSQL($table_name));
+		}
+		catch (Exception $e)
+		{
+			return;
```
Comment (rubencm) [inline phpBB/phpbb/db/tools/doctrine.php:377]: Just ignore the error?

**Comment by rubencm** [comment]
And `fulltext_mysql.php`?

**Comment by marc1706** [comment]
> And `fulltext_mysql.php`, `fulltext_postgres.php`, `fulltext_sphinx.php`?

We could also add a fallback with DELETE FROM for the table in case of an error? That should also work fine but will probably be slower.

File: phpBB/phpbb/db/tools/doctrine.php (hunk 1 above)
Comment (rubencm) [reply phpBB/phpbb/db/tools/doctrine.php:377]: No, I was thinking more in throwing an exception

**Review by rubencm — APPROVED** [review]

**Pull request merged**